### PR TITLE
Add Vernier's GDX-FOR extension

### PR DESCRIPTION
**Pull request by spectranaut**
### Proposed Changes

This PR adds the GDX-FOR extension and 7 initial blocks.

### How to test

You can play with the extension here: https://vernierst.github.io/scratch-gui/goforce/

Make sure you have the most recent Scratch-Link (1.1.28) and a vernier GDX-FOR device.

### CODE REVIEW

File: src/extensions/scratch3_gdx_for/index.js [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+
+/**
+ * Enum for Vernier godirect protocol.
+ * @readonly
+ * @enum {string}
+ */
+const BLEUUID = {
```
Comment (ericrosenbaum) [inline src/extensions/scratch3_gdx_for/index.js:23]: Looks like these UUIDs are also included in the device adapter class- one of them is unused probably?

File: src/extensions/scratch3_gdx_for/index.js [hunk 2]
```diff
@@ -0,0 +1,640 @@
...
+         * The id of the extension this peripheral belongs to.
+         */
+        this._extensionId = extensionId;
+
+        /**
+         * The most recently received value for each sensor.
+         * @type {Object.<string, number>}
+         * @private
+         */
+        this._sensors = {
```
Comment (ericrosenbaum) [inline src/extensions/scratch3_gdx_for/index.js]: looks like this is unused?

File: src/extensions/scratch3_gdx_for/index.js [hunk 3]
```diff
@@ -0,0 +1,640 @@
...
+
+    /**
+     * Enable and begin reading measurements
+     * @private
+     */
+    _startMeasurements () {
+        this._device.sensors.forEach(sensor => {
+            sensor.setEnabled(true);
+            sensor.on('value-changed', changedSensor => {
+                if (changedSensor.values.length > 1000) {
```
Comment (ericrosenbaum) [inline src/extensions/scratch3_gdx_for/index.js:147]: It looks like the vernier module keeps a history of each sensor, and this is clearing it periodically? maybe a comment here could help

File: src/extensions/scratch3_gdx_for/index.js [hunk 4]
```diff
@@ -0,0 +1,640 @@
...
+    }
+
+
+    getForce () {
+        if (this.isConnected()) {
+            let force = this._device.getSensor(1).value;
+            // Normalize the force, which can be measured between -50 and 50 N,
+            // to be a value between -100 and 100.
+            force = force * 2;
+            if (force > 100) {
```
Comment (ericrosenbaum) [inline src/extensions/scratch3_gdx_for/index.js]: we use util.clamp() in other places for things like this

File: src/extensions/scratch3_gdx_for/index.js (hunk 1 above)
Comment (spectranaut) [reply src/extensions/scratch3_gdx_for/index.js:23]: thanks for reminding me about this, I knew I'd have to get back to fixing it when I put them both places XD

File: src/extensions/scratch3_gdx_for/index.js (hunk 3 above)
Comment (spectranaut) [reply src/extensions/scratch3_gdx_for/index.js:147]: I added a comment. Hopefully the new release of @vernier will come out today or tomorrow, this code won't be necessary then!

**Comment by spectranaut** [comment]
I responded to all your comments @ericrosenbaum! I'm working on the rest of the blocks now. So once this is merged I can do a PR with those.

**Review by ericrosenbaum — APPROVED** [review]
LGTM!

**Pull request merged**